### PR TITLE
Add lifecycle end-to-end test with column filters

### PR DESCRIPTION
**Pull request by acco**

### CODE REVIEW

**Comment by acco** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/114?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#114** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/114?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#113** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/113?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#112** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/112?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#111** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/111?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#110** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/110?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @acco and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: test/sequin/postgres_replication_test.exs [hunk 1]
```diff
@@ -298,6 +297,58 @@ defmodule Sequin.PostgresReplicationTest do
...
+
+    test "consumer with column filter only receives relevant events", %{consumer: consumer} do
+      # Create a consumer with a column filter
+      source_tables = [
+        ConsumersFactory.source_table_attrs(
+          oid: Character.table_oid(),
+          actions: [:insert, :update],
+          column_filters: [
+            ConsumersFactory.column_filter_attrs(
+              # Assuming 'name' is the second column
```
Comment (acco) [inline test/sequin/postgres_replication_test.exs:309]: will remove, this is an erroneous comment

File: lib/sequin/extensions/replication.ex [hunk 2]
```diff
@@ -158,6 +162,14 @@ defmodule Sequin.Extensions.Replication do
     {:noreply, state}
   end
 
+  @impl Postgrex.ReplicationConnection
+  def handle_call({:update_message_handler_ctx, ctx}, from, state) do
+    state = %{state | message_handler_ctx: ctx}
+    # Need to manually send reply
+    GenServer.reply(from, :ok)
+    {:noreply, state}
+  end
```
Comment (acco) [inline lib/sequin/extensions/replication.ex:171]: keep the genserver ignorant about this stuff by not reloading and instead sending in?

PostgresReplication doesn't let you reply, so you have to do it.

File: test/support/models/character.ex [hunk 3]
```diff
@@ -16,6 +16,19 @@ defmodule Sequin.Test.Support.Models.Character do
...
   end
 
+  def column_attnums do
+    from(pg in "pg_attribute",
+      where: pg.attrelid == fragment("'characters'::regclass") and pg.attnum > 0,
+      select: {pg.attname, pg.attnum}
+    )
+    |> Sequin.Repo.all()
+    |> Map.new()
+  end
```
Comment (acco) [inline test/support/models/character.ex:26]: i'm digging schemaless - claude makes stuff like this approachable

File: test/sequin/postgres_replication_test.exs (hunk 1 above)
Comment (RTLS) [reply test/sequin/postgres_replication_test.exs:309]: just a reminder

File: test/support/models/character.ex [hunk 4]
```diff
@@ -16,6 +16,19 @@ defmodule Sequin.Test.Support.Models.Character do
     from(c in query, where: c.id == ^id)
   end
 
+  def column_attnums do
+    from(pg in "pg_attribute",
+      where: pg.attrelid == fragment("'characters'::regclass") and pg.attnum > 0,
```
Comment (RTLS) [inline test/support/models/character.ex:21]: why is this filter required? `pg.attnum > 0`

Comment (RTLS) [reply test/support/models/character.ex:21]: looks like it filters out system columns. perhaps leave a comment? claude understands so probably fine lol

**Review by RTLS — APPROVED** [review]

File: lib/sequin/replication_runtime/supervisor.ex [hunk 5]
```diff
@@ -39,27 +39,46 @@ defmodule Sequin.ReplicationRuntime.Supervisor do
...
+    case Sequin.Replication.get_pg_replication(id) do
+      {:ok, pg_replication} ->
+        # Remove races by locking - better way?
+        :global.trans(
+          {__MODULE__, id},
+          fn ->
+            new_ctx = MessageHandler.context(pg_replication)
+            ReplicationExt.update_message_handler_ctx(id, new_ctx)
+          end,
+          [node() | Node.list()],
```
[the remark this answers is not in the document]
Comment (acco) [reply lib/sequin/replication_runtime/supervisor.ex:59]: knew you'd like

File: lib/sequin/replication_runtime/supervisor.ex [hunk 6]
```diff
@@ -39,27 +39,46 @@ defmodule Sequin.ReplicationRuntime.Supervisor do
...
+          5000
```
[the remark this answers is not in the document]
Comment (acco) [reply lib/sequin/replication_runtime/supervisor.ex:60]: Ah! Seems like we can't really set a timeout, so I'll set it to 20 -- it's such an edge case, and should resolve quickly

File: test/support/models/character.ex (hunk 4 above)
Comment (acco) [reply test/support/models/character.ex:21]: Leaving comment
